### PR TITLE
smtp mail support

### PR DESCRIPTION
**Pull request by m-adawi**

### CODE REVIEW

File: src/main/java/org/otherband/email/simulation/EmailServiceSimulator.java [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
+import org.otherband.Utils;
 import org.springframework.context.annotation.Profile;
 import org.springframework.stereotype.Component;
 import org.otherband.email.EmailService;
 import org.otherband.email.VerificationLinkEmailRequest;
-import org.otherband.letter.RecommendationLetterController;
 
 import java.util.UUID;
 
 import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.linkTo;
```
Comment (osama-salman99) [inline src/main/java/org/otherband/email/simulation/EmailServiceSimulator.java:11]: Remove unused import

File: src/main/java/org/otherband/Utils.java [hunk 2]
```diff
@@ -11,4 +15,11 @@ private Utils() {
     public static void validateNotBlank(String field, String fieldName) {
         if (isBlank(field)) throw UserInputException.formatted("[%s] cannot be blank", fieldName);
     }
+
+    public static String buildLetterVerificationLink(VerificationLinkEmailRequest verificationLinkEmailRequest) {
+        return linkTo(methodOn(RecommendationLetterController.class)
+                .verify(verificationLinkEmailRequest.getLetterId(),
+                        verificationLinkEmailRequest.getTokenId(),
+                        verificationLinkEmailRequest.getSecretToken())).withSelfRel().toString();
+    }
```
Comment (osama-salman99) [inline src/main/java/org/otherband/Utils.java:24]: Can we move this back to EmailServiceSimulator.java as this method is not for general usage and is strictly related to the EmailService

File: src/test/java/org/otherband/email/smtp/SMTPEmailServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        mailSender = mock(JavaMailSender.class);
+        service = new SMTPEmailService(mailUserName, mailSender);
+    }
+
+    @Test
+    void testSendVerificationEmail() {
+        VerificationLinkEmailRequest verificationLinkEmailRequest = new VerificationLinkEmailRequest();
+        service.sendLetterVerificationLink(verificationLinkEmailRequest);
+        verify(mailSender).send(Mockito.any(SimpleMailMessage.class));
+    }
```
Comment (osama-salman99) [inline src/test/java/org/otherband/email/smtp/SMTPEmailServiceTest.java:29]: Can we add content to the VerificationLinkEmailRequest and verify the fields of the created SimpleMailMessage

File: src/main/java/org/otherband/email/simulation/EmailServiceSimulator.java (hunk 1 above)
Comment (m-adawi) [reply src/main/java/org/otherband/email/simulation/EmailServiceSimulator.java]: done

File: src/main/java/org/otherband/Utils.java (hunk 2 above)
Comment (m-adawi) [reply src/main/java/org/otherband/Utils.java:24]: Yeah but we have multiple implementations of EmailService interface so we need it in multiple places. I was thinking of making it part of the VerificationLinkEmailRequest class but @yazan-abu-obaideh said we shouldn't add methods to DTO classes so we decided to put it in Utils

File: src/test/java/org/otherband/email/smtp/SMTPEmailServiceTest.java (hunk 3 above)
Comment (m-adawi) [reply src/test/java/org/otherband/email/smtp/SMTPEmailServiceTest.java:29]: How do I access the created SimpleMailMessage in this context when it's local to the sendLetterVerificationLink method? should we have the method return it?

File: src/main/java/org/otherband/Utils.java (hunk 2 above)
Comment (osama-salman99) [reply src/main/java/org/otherband/Utils.java:24]: Makes absolute sense 💯

File: src/test/java/org/otherband/email/smtp/SMTPEmailServiceTest.java (hunk 3 above)
Comment (osama-salman99) [reply src/test/java/org/otherband/email/smtp/SMTPEmailServiceTest.java:29]: You can use Mockito's ArgumentCaptor to capture the SimpleMailMessage

Comment (yazan-abu-obaideh) [reply src/test/java/org/otherband/email/smtp/SMTPEmailServiceTest.java:29]: resolved

**Review by yazan-abu-obaideh — APPROVED** [review]

**Review by yazan-abu-obaideh — APPROVED** [review]

**Review by osama-salman99 — APPROVED** [review]

**Pull request merged**